### ISSUE TITLE
Include Abyss Finance deposits to solo stakers.

### ISSUE THREAD

**Issue by geomad** [GLCNI/ETH-Solo-Validator-Addresses#10]
`0xfa5f9eaa65ffb2a75de092eb7f3fc84fc86b5b18` is labeled as `Abyss Finance` in `Entity-list.csv`.
This smart contract is used by users to deposit multiple validators, with just one transaction.

Quoting from `https://abyss.finance/` :

```
Ethereum Depositor
Abyss Ethereum Depositor is a convenient tool that will help you to make multiple deposits to Ethereum Deposit Contract in one batch.
```

Deposits from Abyss Finance should also be classified, as multiple deposits are solo stakers that use this smart contract for convenience.

**Comment by GLCNI** [comment]
Hey, thank you for pointing this out. I excluded contract address deposits as an easy way to filter out entities, except for Rocketpool as they account for such a large number of solo stakers on the network.

I wasn't aware of this tool at the time, now that I am, I think its worth adding the addresses that interacted with the contract (that are non entities), any help would be appreciated on how to add them.

**Comment by geomad** [comment]
> Hey, thank you for pointing this out. I excluded contract address deposits as an easy way to filter out entities, except for Rocketpool as they account for such a large number of solo stakers on the network.
> 
> I wasn't aware of this tool at the time, now that I am, I think its worth adding the addresses that interacted with the contract (that are non entities), any help would be appreciated on how to add them.

Thank you!  I'm not familiar with Dune but if looking on chain with an archive node an easy way, would be to filter the transactions that have `transaction.to` set to the Abyss Finance address, called the `deposit` method and were not reverted. The `transaction.from` field should be the EOA that made the deposit.

This method though, would exclude any multisig/smart contract wallets. If you want to include those then I think every tx on chain needs to be traced, and looks for calls on the Abyss Finance contract, and the origin of the call.

Using Dune or cryo should be a lot faster though!

Please let me know what you think, I could write the code for that, and even run the queries in my node.

**Comment by geomad** [comment]
So I played around with DUNE a bit and it looks like this query works on finding both the EOAs and the smart contracts which interacted with abyss finance, which in turn interacted with the deposit contract. It makes sure both the call from abyss to the deposit contract and the transaction were successful.
```
WITH RelevantCalls AS (
  SELECT
    "from",
    "to",
    "tx_hash",
    "block_time",
    "success",
    "tx_success"
  FROM
    ethereum."traces"
  WHERE
    ("to" = 0xFA5F9EAA65FFB2A75DE092EB7F3FC84FC86B5B18 OR
     "from" = 0xFA5F9EAA65FFB2A75DE092EB7F3FC84FC86B5B18) AND
    "error" IS NULL AND
    "block_time" >= CAST('2020-12-01' AS timestamp) AND
    "success" = true AND
    "tx_success" = true
)
SELECT DISTINCT
  rc1."from" AS original_caller
FROM
  RelevantCalls rc1
JOIN
  RelevantCalls rc2
ON
  rc1."tx_hash" = rc2."tx_hash" AND
  rc1."to" = 0xFA5F9EAA65FFB2A75DE092EB7F3FC84FC86B5B18 AND
  rc2."from" = 0xFA5F9EAA65FFB2A75DE092EB7F3FC84FC86B5B18 AND
  rc2."to" = 0x00000000219AB540356CBB839CBE05303D7705FA
```

**Comment by geomad** [comment]
Hey @GLCNI any plans to fix this? We keep getting omitted from drops, that feels bad! We got a soul too!

**Comment by Stake-Cat** [comment]
Hey @geomad this is to be added in an update I'm planning to push tonight, which includes up to dencun hardfork

**Comment by Stake-Cat** [comment]
Hi @geomad 
thanks for your help so far, couldn’t include this in the last update.

Using your query it looks like **1808 unique addresses,** and looking at the activity most profiles do not look like Solo Staker behaviour.
_who uses Abyss: it seems like the occasional solo staker, people looking to deposit many multiple validators usually on behalf of someone else, ie entities._

The reason for excluding deposits from contract addresses is a filter, as its very unlikely that Solo Stakers would deposit via contract. 

I understand that this tool is used by some solo stakers, and they should be included, but we need to find a way to identify the Solo Stakers within this set. If you have any ideas it would be appreciated.

**Comment by geomad** [comment]
> Hi @geomad thanks for your help so far, couldn’t include this in the last update.
> 
> Using your query it looks like **1808 unique addresses,** and looking at the activity most profiles do not look like Solo Staker behaviour. _who uses Abyss: it seems like the occasional solo staker, people looking to deposit many multiple validators usually on behalf of someone else, ie entities._
> 
> The reason for excluding deposits from contract addresses is a filter, as its very unlikely that Solo Stakers would deposit via contract.
> 
> I understand that this tool is used by some solo stakers, and they should be included, but we need to find a way to identify the Solo Stakers within this set. If you have any ideas it would be appreciated.

Solo stakers, including my self are definitely using this! Sending multiple transactions to stake multiple validators is very time consuming (especially when a hardware wallet is involved), and also wastes a lot more gas!

Filtering out people using this smart contract just because they used this, makes no sense, since entities could have deposited directly to the beacon contract as well, and they would get included in this snapshot. Excluding unknown entites should be done in a different way (maybe by analyzing onchain actions? Not sure how that would help either though) for both cases.

A filter that can be added in my query is to query only for EOAs that interacted with Abyss contract. This would exclude protocols that use smart contracts to manage the funds, but would also exclude people that staked through smart wallets or multisigs. I'm not sure which approach you took for this matter in the original query.

**Comment by ksheni56** [comment]
I also used Abyss Finance Contract because it was easy to use, but still not in the Solo Staker list. Hope we can get in there soon?

**Comment by geomad** [comment]
any plans on taking some action on this? Another airdrop was lost today.

**Comment by ksheni56** [comment]
Yeah so bad. Would it please be possible for the small part of the community that doesn't want to be forced to restart their Allnodes (whatever) nodes to use a custom label graffiti so that they can participate in airdrops?
